### PR TITLE
Multiple uploaders in the same page don't work

### LINKED ISSUE #2307: Multiple uploaders in the same page don't work
Input fields of the uploader component don’t have an unique ID.

As a consequence, if multiple uploaders are present on the same page, when the user clicks the “browse” button then the file will always be uploaded using the first one.
Drag & drop upload is not affected by this bug.

Drag & drop upload is not affected by this bug.

### PR DESCRIPTION
**Pull request by davide-negretti**
## References
* Fixes #2307

## Description
The input field of the file uploader was missing an unique ID, causing issues when multiple uploaders are used on the same page

## Instructions for Reviewers

To reproduce the bug:
 - add two or more uploaders to a test component
 - drag and drop works fine on both uploader
 - if you click "browse" button on the second uploader the file will be uploaded using the first one

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

File: src/app/shared/upload/uploader/uploader.component.html [hunk 1]
```diff
@@ -22,10 +22,10 @@
         <span>
           <i class="fas fa-upload" aria-hidden="true"></i>
           {{dropMsg | translate}}{{'uploader.or' | translate}}
-          <label for="inputFileUploader" class="btn btn-link m-0 p-0 ml-1" tabindex="0" (keyup.enter)="$event.stopImmediatePropagation(); fileInput.click()">
+          <label for="inputFileUploader{{uploaderId}}" class="btn btn-link m-0 p-0 ml-1" tabindex="0" (keyup.enter)="$event.stopImmediatePropagation(); fileInput.click()">
             <span role="button" [attr.aria-label]="'uploader.browse' | translate">{{'uploader.browse' | translate}}</span>
           </label>
-          <input #fileInput id="inputFileUploader" class="d-none" type="file" ng2FileSelect [uploader]="uploader" multiple tabindex="0" />
+          <input #fileInput id="inputFileUploader{{uploaderId}}" class="d-none" type="file" ng2FileSelect [uploader]="uploader" multiple tabindex="0" />
```
Comment (tdonohue) [inline src/app/shared/upload/uploader/uploader.component.html:28]: While this works... the `uploaderId` appears to always be `ds-drag-and-drop-uploader[unique-id]`.  So, the final ID looks like this: `inputFileUploaderds-drag-and-drop-uploader[unique-id]`.

I wonder if we should add a dash here to be: `id="inputFileUploader-{{uploaderId}}"`... that makes it a bit of a more understandable ID.

**Review by tdonohue — COMMENTED** [review]
@davide-negretti : Overall this works.  But, I wonder if we should improve this `id` slightly so that it doesn't appear all squished together.  See inline comment below.

Either way, I'm +1 this PR.  But, wanted to note that the currently created `id` is very hard to read/understand (even though it is unique).

File: src/app/shared/upload/uploader/uploader.component.html (hunk 1 above)
Comment (davide-negretti) [reply src/app/shared/upload/uploader/uploader.component.html]: @tdonohue I added the missing dash

**Comment by davide-negretti** [comment]
@tdonohue I added the missing dash

**Review by tdonohue — APPROVED** [review]
👍 Thanks @davide-negretti !  Looks good to me now

**Pull request merged**